driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -635,14 +639,14 @@ public void shutdown() {
             // we're not on an I/O thread or anything, so just call await.
             allChannels.close().awaitUninterruptibly();
 
-            // This will call shutdownNow on the boss and worker executor. Since this is called
-            // only once all connection have been individually closed, it's fine.
-            channelFactory.releaseExternalResources();
+            // This will shut down I/O threads. Since this is called only once all connection have
+            // been individually closed, it's fine.
+            eventLoopGroup.shutdownGracefully().syncUninterruptibly();
```

Comment:
> won't we not want to do this if the eventLoopGroup was passed in from the outside? 

That's a very good point and I'm going to add a comment on JAVA-676 to remember that.

> We could also consider sharing EventLoopGroup between all Cluster instances

I think this would be a bad idea, and if people ever want to do that they will have JAVA-676.